src/video/kmsdrm/SDL_kmsdrmvideo.c

```diff
@@ -75,37 +72,40 @@ static int
 get_driindex(void)
 {
     int available = -ENOENT;
-    char device[512] = KMSDRM_DRI_PATH;
+    char device[32];
     int drm_fd;
     int i;
     int devindex = -1;
-    DIR *folder = opendir(device);
+    DIR *folder;
+
+    SDL_strlcpy(device, kmsdrm_dri_path, kmsdrm_dri_pathsize + 1);
+    folder = opendir(device);
     if (!folder) {
         SDL_SetError("Failed to open directory '%s'", device);
         return -ENOENT;
     }
 
-    strncpy(device + KMSDRM_DRI_PATHSIZE, KMSDRM_DRI_DEVNAME,
-            KMSDRM_DRI_DEVNAMESIZE + 1);
+    SDL_strlcpy(device + kmsdrm_dri_pathsize, kmsdrm_dri_devname,
```

Comment:
Again here, this should be `sizeof(device) - kmsdrm_dri_pathsize`

Reply:
kmsdrm_dri_cardpath rather, but yes I can do this:

`char device[sizeof(kmsdrm_dri_cardpath)];`